app/src/ui/dialog/dialog.tsx

```diff
@@ -187,8 +187,12 @@ export class Dialog extends React.Component<IDialogProps, IDialogState> {
     }
 
     if (this.props.title) {
+      // createUniqueId handles static strings fine, so in the case of receiving
+      // a JSX element for the title we can just pass in a fixed value rather
+      // than trying to generate a string from an arbitrary element
+      const id = typeof this.props.title === 'string' ? this.props.title : '???'
```

Comment:
Not gonna hold up this from landing because of this but `???` stands out to me here. I would have preferred if we either left this code as it was, or that we made this constant something like `JSX` rather than triple question marks. It's entirely stylistic though so I'm gonna go ahead and approve and land this.